### PR TITLE
DrivAerML: 4L/256d+T_max=30 — lr=6e-4 (mid-point higher LR)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

lr=5e-4 is the current best for 4L/256d+T_max=30 DrivAerML (12.70% val). Testing lr=6e-4 (between 5e-4 and 8e-4) to probe whether a higher LR can squeeze out additional performance. This is part of a fine-grained LR sweep (alongside lr=4e-4 in a parallel PR). If LR has a peak in this range, we want to locate it precisely.

## Instructions

Run the following command exactly:

```bash
cd target/icml2026 && python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 6e-4 \
  --cosine_t_max 30 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --wandb_group drivaerml-4L256d-lr-sweep
```

**CRITICAL OOM flags — do not omit:** `--batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 --model-heads 4`

## Baseline

- **Dataset:** DrivAerML
- **Primary metric:** `val_primary/surface_rel_l2_pct` (lower is better)
- **Current best val:** 12.70%
- **Current best test:** 13.54%
- **Best PR:** #2593 (shinji — Fourier+4L/256d+no-EMA+T_max=30, 45 epochs, AdamW lr=5e-4)
- **W&B run:** 3aaevlho
- **Baseline reproduce:** `cd target/icml2026 && python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine_t_max 30 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 256 --model-heads 4 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200`

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT violet:

## Results

### lr=6e-4 (DrivAerML 4L/256d+T_max=30)

| Metric | Baseline (lr=5e-4) | This run (lr=6e-4) | Delta |
|--------|-------------------|---------------------|-------|
| `val_primary/surface_rel_l2_pct` | **12.70%** | 13.92% | +1.22% (worse) |
| `test_primary/surface_rel_l2_pct` | **13.54%** | 14.75% | +1.21% (worse) |

- **Best val:** 13.92% at epoch 40
- **Final val:** 14.04% at epoch 45
- **Test:** 14.75% (evaluated at final epoch)
- **Epochs completed:** 45 (hit 30-min timeout)
- **W&B run:** `nojfzj9i` ([link](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/nojfzj9i))
- **W&B group:** `drivaerml-4L256d-lr-sweep`

**Command used:**
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset drivaerml --optimizer adamw --lr 6e-4 --cosine-t-max 30 \
  --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 256 \
  --model-heads 4 --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --wandb-group drivaerml-4L256d-lr-sweep \
  --agent violet --wandb-name "violet/drivaerml-lr6e4"
```

### What happened

**lr=6e-4 does not beat the baseline lr=5e-4.** The higher learning rate resulted in ~1.2% worse performance on both val and test metrics. The model converged steadily through 45 epochs with good loss reduction (0.81 → 0.039), but validation metrics plateaued higher than the baseline.

The convergence curve showed the expected cosine annealing pattern — best val scores appeared near LR troughs (epochs 38, 40). However, even at these optimal points, the model couldn't match the 12.70% achieved by lr=5e-4.

This suggests **lr=5e-4 is closer to optimal than lr=6e-4** for this architecture. The LR response curve appears to favor the lower side of the 5e-4 range.

### Suggested follow-ups

- Combined with the lr=4e-4 parallel PR results, this should clarify whether 5e-4 is a local optimum or if the sweet spot is slightly below 5e-4 (e.g., 4.5e-4).
- A finer sweep between 4e-4 and 5e-4 could be worthwhile if lr=4e-4 also underperforms.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2625 because ADVISOR: Dead end — DrivAerML lr=6e-4 reached 13.42% (vs baseline now 11.97%). LR landscape fully mapped: lr=5e-4 remains optimal. Closing..
